Project.toml

```diff
@@ -27,6 +27,7 @@ OrderedCollections = "bac558e1-5e72-5ebc-8fee-abe8a469f55d"
 Pkg = "44cfe95a-1eb2-52ea-b672-e2afdf69b78f"
 Printf = "de0858da-6303-5e67-8744-51eddeeeb8d7"
 Random = "9a3f8284-a2c9-5f02-9a11-845980a1fd5c"
+ReactantCore = "a3311ec8-5e00-46d5-b541-4f83e724a433"
```

Comment:
do we need this as main dependency or just weak dependency?

Reply:
Yes, because it's used in the main package, not an extension.  Note that `ReactantCore` is a much smaller dependency than `Reactant` itself, [it's fairly small](https://github.com/EnzymeAD/Reactant.jl/blob/main/lib/ReactantCore/src/ReactantCore.jl) and [it has only two dependencies](https://github.com/EnzymeAD/Reactant.jl/blob/0e7287bd1cf543313c0d48f702cce009aaa2a952/lib/ReactantCore/Project.toml#L6), chances are Oceananigans already indirectly depended on at least one of them through other packages.

Reply:
No worries -- great!
Just wanted to ensure that it wasn't there by accident ;)